UM/Math/Matrix.py

```diff
@@ -140,6 +168,177 @@ def setByRotationAxis(self, angle, direction, point = None):
+            R = Matrix()
+            R.setByEuler(angles.x, angles.y, angles.z, 'sxyz')
+            M = numpy.dot(M, R.getData())
+        if shear is not None:
+            Z = numpy.identity(4)
+            Z[1, 2] = shear.x
+            Z[0, 2] = shear.y
+            Z[0, 1] = shear.z
+            M = numpy.dot(M, Z)
+        if scale is not None:
+            S = numpy.identity(4)
+            S[0, 0] = scale.x
+            S[1, 1] = scale.y
+            S[2, 2] = scale.z
+            M = numpy.dot(M, S)
+        M /= M[3, 3]
+        self._data = M
+
+    def getEuler(self, axes='sxyz'):
+        """Return Euler angles from rotation matrix for specified axis sequence.
+
+        axes : One of 24 axis sequences as string or encoded tuple
+
+        Note that many Euler angle triplets can describe one matrix.
+
+        >>> R0 = euler_matrix(1, 2, 3, 'syxz')
+        >>> al, be, ga = euler_from_matrix(R0, 'syxz')
+        >>> R1 = euler_matrix(al, be, ga, 'syxz')
+        >>> numpy.allclose(R0, R1)
+        True
+        >>> angles = (4*math.pi) * (numpy.random.random(3) - 0.5)
+        >>> for axes in _AXES2TUPLE.keys():
+        ...    R0 = euler_matrix(axes=axes, *angles)
+        ...    R1 = euler_matrix(axes=axes, *euler_from_matrix(R0, axes))
+        ...    if not numpy.allclose(R0, R1): print(axes, "failed")
+
+        """
+        try:
+            firstaxis, parity, repetition, frame = self._AXES2TUPLE[axes.lower()]
+        except (AttributeError, KeyError):
+            self._TUPLE2AXES[axes]  # validation
+            firstaxis, parity, repetition, frame = axes
+
+        i = firstaxis
+        j = self._NEXT_AXIS[i + parity]
+        k = self._NEXT_AXIS[i - parity + 1]
+
+        M = numpy.array(self._data, dtype = numpy.float64, copy = False)[:3, :3]
+        if repetition:
+            sy = math.sqrt(M[i, j] * M[i, j] + M[i, k] * M[i, k])
+            if sy > self._EPS:
+                ax = math.atan2( M[i, j],  M[i, k])
+                ay = math.atan2( sy,       M[i, i])
+                az = math.atan2( M[j, i], -M[k, i])
+            else:
+                ax = math.atan2(-M[j, k],  M[j, j])
+                ay = math.atan2( sy,       M[i, i])
+                az = 0.0
+        else:
+            cy = math.sqrt(M[i, i] * M[i, i] + M[j, i] * M[j, i])
+            if cy > self._EPS:
+                ax = math.atan2( M[k, j],  M[k, k])
+                ay = math.atan2(-M[k, i],  cy)
+                az = math.atan2( M[j, i],  M[i, i])
+            else:
+                ax = math.atan2(-M[j, k],  M[j, j])
+                ay = math.atan2(-M[k, i],  cy)
+                az = 0.0
+
+        if parity:
+            ax, ay, az = -ax, -ay, -az
+        if frame:
+            ax, az = az, ax
+        return Vector(ax, ay, az)
+
+
+    def setByEuler(self, ai, aj, ak, axes = "sxyz"):
+        """Return homogeneous rotation matrix from Euler angles and axis sequence.
+
+        ai, aj, ak : Euler's roll, pitch and yaw angles
```

Comment:
I'd prefer just to call them roll, pitch and jaw then, but maybe there's some merit to keeping this code aligned with Gohlke's code too. Something to consider.